src/librustc/ty/maps.rs

```diff
@@ -580,10 +598,17 @@ macro_rules! define_maps {
                        key,
                        span);
 
+                profq_msg!(tcx,
+                    ProfileQueriesMsg::QueryBegin(span.clone(),
+                                                  QueryMsg::$name(format!("{:?}", key))));
```

Comment:
Out of curiosity, is your concern here with the time required to format the key, the space required to store it (or time/space to write it out to output files), or all of the above?